File: userapi/src/main/java/com/teleri/userapi/dto/Location.java
package com.teleri.userapi.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Location {
    private Street street;
    private String city;
    private String state;
    private String country;
    private String postcode;
    private Coordinates coordinates;
    private Timezone timezone;
}

File: userfront/src/app/models/location.ts
import { Coordinates } from "./coordinates";
import { Street } from "./street";
import { Timezone } from "./timezone";

export class Location {

    street: Street;
    city: string;
    state: string;
    country: string;
    postcode: string;
    coordinates: Coordinates;
    timezone: Timezone;

    constructor(
        street: Street,
        city: string,
        state: string,
        country: string,
        postcode: string,
        coordinates: Coordinates,
        timezone: Timezone
    ) {
        this.street = street
        this.city = city
        this.state = state
        this.country = country
        this.postcode = postcode
        this.coordinates = coordinates
        this.timezone = timezone
    }

}
